src/main/java/com/epam/ta/reportportal/core/project/impl/DeleteProjectHandlerImpl.java

```diff
@@ -96,6 +102,10 @@ public OperationCompletionRS deleteProjectIndex(String projectName, String usern
 				Predicate.isEqual(false)
 		).verify(ErrorType.FORBIDDEN_OPERATION, "Index can not be removed until index generation proceeds.");
 
+		expect(analyzerServiceClient.hasClients(), Predicate.isEqual(true)).verify(ErrorType.UNABLE_INTERACT_WITH_INTEGRATION,
```

Comment:
it is a pretty strong condition, so you can move it to the beginning of the method to avoid logic of project and user retrieving.